Ullaakut/Bloggo
# Add token signature verification & unit tests 100% (#51) (#52)

<img width="949" alt="screenshot 2018-08-05 at 22 04 15" src="https://user-images.githubusercontent.com/6976628/43689473-85f274ee-98fb-11e8-9f1d-b4a67e0c603d.png">

## Goal of this PR

Fixes #51 
Fixes #52 

This PR is kind of a last minute fix on a few things I realized I completely forgot.

This token signatures were not verified, and when adding some features previously, I reduced the unit test coverage and forgot to check it again before testing.

That wouldn't have happened with a coveralls integration, or a git hook running the unit tests, so I feel pretty dumb about it, but at least it's fixed now!

- [x] Update postman collection token
- [x] Update README tokens
- [x] Update unit test tokens
- [x] Update documentation to explain that the secret should be set when creating your own token

## How to test it

* Take the valid token from the `README.md` file and put it in [jwt.io](https://jwt.io/)
* Change its signature and copy the new token (this is just to make sure that the signature is a valid string but not the right one)
* Try to use this token for your API calls
* It should fail

* Take the valid token from the `README.md` file
* Try to use this token for your API calls
* It should succeed

* Run `go test controller/*.go -cover`
* Test coverage should be 100%

* Run `go test service/*.go -cover`
* Test coverage should be 100%

<img width="685" alt="screenshot 2018-08-05 at 22 03 00" src="https://user-images.githubusercontent.com/6976628/43689458-5853c8a8-98fb-11e8-83d8-8e2806fcb76e.png">
